From 3827ced92e9df5168499cdc7fc5e4eaffc2d5819 Mon Sep 17 00:00:00 2001
From: Linu Cherian <linu.cherian@cavium.com>
Date: Mon, 20 Aug 2018 15:05:08 +0530
Subject: [PATCH 0149/1051] soc: octeontx2: Fix irq cleanup in cgx driver
 remove path

Free the irq resources on driver remove.
Along with that fix a debug print as well.

Signed-off-by: Nithya <nmani@marvell.com>
Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/cgx.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2/cgx.c b/drivers/soc/marvell/octeontx2/cgx.c
index 5b5f7cc62434..8c24e3e5bc28 100644
--- a/drivers/soc/marvell/octeontx2/cgx.c
+++ b/drivers/soc/marvell/octeontx2/cgx.c
@@ -466,7 +466,7 @@ static inline void cgx_link_change_handler(struct cgx_lnk_sts *lstat,
 		dev_dbg(dev, "cgx port %d:%d Link change handler null",
 			cgx->cgx_id, lmac->lmac_id);
 		if (lstat->err_type != CGX_ERR_NONE) {
-			dev_err(dev, "cgx port %d:%d Link error %x\n",
+			dev_err(dev, "cgx port %d:%d Link error %d\n",
 				cgx->cgx_id, lmac->lmac_id, lstat->err_type);
 		}
 		dev_info(dev, "cgx port %d:%d Link is %s %d Mbps\n",
@@ -669,6 +669,7 @@ static int cgx_lmac_exit(struct cgx *cgx)
 		lmac = cgx->lmac_idmap[i];
 		if (!lmac)
 			continue;
+		free_irq(pci_irq_vector(cgx->pdev, CGX_LMAC_FWI + i * 9), lmac);
 		kfree(lmac->name);
 		kfree(lmac);
 	}
@@ -748,6 +749,7 @@ static void cgx_remove(struct pci_dev *pdev)
 
 	cgx_lmac_exit(cgx);
 	list_del(&cgx->cgx_list);
+	pci_free_irq_vectors(pdev);
 	pci_release_regions(pdev);
 	pci_disable_device(pdev);
 	pci_set_drvdata(pdev, NULL);
-- 
2.17.1

